src/entity/task/new_task.rs

```diff
@@ -0,0 +1,135 @@
+use super::*;
+
+const URL_SAFE_BASE64_CHARS: &[u8] =
+    b"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
+
+fn url_safe_base64(data: &str) -> Result<(), ValidationError> {
+    if data
+        .chars()
+        .all(|c| URL_SAFE_BASE64_CHARS.contains(&(c as u8)))
```

Comment:
nit: while this validation is good enough for practical purposes, truncating instead of using `try_into()` here will incorrectly allow, for example, "MEDIUM MATHEMATICAL SPACE (U+205F)"

Reply:
Say more? I don't understand the concern here

Reply:
Is it the `c as u8`? In that case `URL_SAFE_BASE64_CHARS` wouldn't include it

Reply:
The `as` conversion truncates from `char` to `u8`, so this validation will incorrectly allow a smattering of non-ASCII Unicode codepoints, so long as the codepoint's lower eight bits happen to match an allowed character. See this [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a2ef1a256b794112ad25fa555fb6acac).

Reply:
Oh yikes! Thanks!

Reply:
resolved in 1561dc1 and test case added